indico_zodbimport/modules/event_paper_reviewing.py

```diff
@@ -0,0 +1,128 @@
+        if reviewing:
+            for resource in reviewing._Material__resources.itervalues():
+                self._migrate_resource(resource, new_contribution, reviewing._modificationDS)
+        for review in old_contribution._reviewManager._versioning:
+            if review._materials:
+                assert len(review._materials) == 1
+                for resource in review._materials[0]._Material__resources.itervalues():
+                    self._migrate_resource(resource, new_contribution, reviewing._modificationDS,
+                                           version=review._version)
+        db.session.commit()
+
+    def migrate_reviewing(self):
+        self.print_step('building new ZODB index for reviewing objects')
+
+        for old_event in self._zodb_committing_iterator(self._iter_events()):
+            self.event_new = old_event.as_event
+            conference_settings = old_event._confPaperReview
+            contrib_index = conference_settings._contribution_index = IOBTree()
+            contrib_reviewers = _invert_mapping(conference_settings._reviewerContribution)
+            contrib_referees = _invert_mapping(conference_settings._refereeContribution)
+            contrib_editors = _invert_mapping(conference_settings._editorContribution)
+
+            for old_contribution in old_event.contributions.itervalues():
+                review_manager = getattr(old_contribution, '_reviewManager')
+                new_contribution = Contribution.find_one(event_id=self.event_new.id,
+                                                         friendly_id=int(old_contribution.id))
+
+                cid = int(new_contribution.id)
+                if not review_manager:
+                    self.print_warning('Contribution {} had no ReviewManager. Fixed.'.format(cid),
+                                       event_id=old_event.id)
+                    contrib_index[cid] = old_contribution._reviewManager = ReviewManager(cid)
+                else:
+                    review_manager._contrib_id = cid
+                    contrib_index[cid] = review_manager
+
+                self._migrate_contribution_roles(old_contribution, new_contribution, contrib_reviewers,
+                                                 PaperReviewingRoleType.reviewer)
+                self._migrate_contribution_roles(old_contribution, new_contribution, contrib_referees,
+                                                 PaperReviewingRoleType.referee)
+                self._migrate_contribution_roles(old_contribution, new_contribution, contrib_editors,
+                                                 PaperReviewingRoleType.editor)
+
+                self._migrate_reviewing_materials(old_contribution, new_contribution)
+
+    def _iter_events(self):
+        old_events = self.zodb_root['conferences']
+        return verbose_iterator([self.zodb_root['conferences']['93877']], 1, attrgetter('id'), attrgetter('title'))
```

Comment:
:warning:

- only a single event (testing leftover i guess)
- title getter (query spam)
- skip events that aren't conferences (check `EventMenuImporter._iter_events`)